### PR TITLE
Fix - Changing column width changes adjacent column sorting order

### LINKED ISSUE #624: [BUG] Resizing table widget width changes the sorting order of adjacent column
## Description

Please check this video to see the issue in action:
https://www.loom.com/share/4655175f1e1e4e8ab9f23b53d1cece73

### Important Details
 - Version: Cloud
 - OS: MacOSX
 - Browser: chrome

### PR DESCRIPTION
**Pull request by vicky-primathon**
## Description
Changing column width changes adjacent column sorting order

Fixes #624 

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
- Manually tested

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/client/src/components/designSystems/appsmith/TableUtilities.tsx [hunk 1]
```diff
@@ -702,7 +703,7 @@ export const TableHeaderCell = (props: {
     toggleRenameColumn(false);
   };
   const handleSortColumn = () => {
-    if (column.isResizing) return;
+    if (props.isResizingColumn) return;
```
Comment (riodeuno) [inline app/client/src/components/designSystems/appsmith/TableUtilities.tsx:706]: We need a way to disable this so that sorting is available.

**Review by riodeuno — CHANGES_REQUESTED** [review]

File: app/client/src/components/designSystems/appsmith/Table.tsx [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import React from "react";
+import React, { useCallback } from "react";
```
Comment (riodeuno) [inline app/client/src/components/designSystems/appsmith/Table.tsx:1]: Please remove this import of `useCallback` as it is unused.

**Review by riodeuno — DISMISSED** [review]

**Comment by Nikhil-Nandagopal** [comment]
Hey @riodeuno @vicky-primathon  I saw the PR for the column width interaction, it still doesn't work very well. When I drag a column to the right to increase it's width, the left also expands and crushes the columns to the left

**Comment by Nikhil-Nandagopal** [comment]
standup_query
There's a problem with the PR where the columns mismatch with the cells

File: app/client/src/components/designSystems/appsmith/ReactTableComponent.tsx [hunk 3]
```diff
@@ -274,19 +273,17 @@ const ReactTableComponent = (props: ReactTableComponentProps) => {
...
-          [column.accessor]: width,
-        };
-    props.handleResizeColumn(columnSizeMap);
-  };
+  // const handleResizeColumn = (columnId: string, columnWidth: number) => {
+  //   const columnSizeMap = props.columnSizeMap
+  //     ? {
+  //         ...props.columnSizeMap,
+  //         [columnId]: columnWidth,
+  //       }
```
Comment (riodeuno) [inline app/client/src/components/designSystems/appsmith/ReactTableComponent.tsx:281]: Please remove this commented set of lines

**Review by riodeuno — APPROVED** [review]

**Review by Nikhil-Nandagopal — CHANGES_REQUESTED** [review]
@vicky-primathon @riodeuno seeing a big issue on this deploy preview where new apps have a broken canvas

**Review by Nikhil-Nandagopal — APPROVED** [review]

**Pull request merged**